harpy/scripts/demultiplex_gen1.py

```
@@ -0,0 +1,169 @@
+                code_letters.add(segment_id[0])
+            except ValueError:
+                # skip rows without two columns
+                continue
+    id_letter = code_letters.pop()
+    return id_letter, data_dict
+
+def get_min_dist(needle, code_letter):
+    minDist = 999
+    nbFound = 0
+    minSeq =""
+    for seq in bar_codes[code_letter].keys():
```

Ruff (0.8.2)

Use `key in dict` instead of `key in dict.keys()`

Remove `.keys()`

(SIM118)